lib/spack/docs/tutorial_advanced_packaging.rst

```diff
@@ -0,0 +1,560 @@
+  ==> Executing phase: 'cmake'
+  ==> Executing phase: 'build'
+  ==> Executing phase: 'install'
+  ==> Successfully installed armadillo
+    Fetch: 0.01s.  Build: 3.96s.  Total: 3.98s.
+  [+] /usr/local/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/armadillo-8.100.1-n2eojtazxbku6g4l5izucwwgnpwz77r4
+
+The installation went fine and the code we added expanded to the right list
+of semicolon separated libraries (you are encouraged to open ``armadillo``'s
+build logs to double check).
+
+If we try to build another version tied to ``netlib-lapack`` we'll
+notice that this time the installation won't complete:
+
+.. code-block:: console
+
+  root@advanced-packaging-tutorial:/# spack install  armadillo ^netlib-lapack
+  ==> pkg-config is already installed in /usr/local/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/pkg-config-0.29.2-ae2hwm7q57byfbxtymts55xppqwk7ecj
+  ...
+  ==> openmpi is already installed in /usr/local/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/openmpi-3.0.0-yo5qkfvumpmgmvlbalqcadu46j5bd52f
+  ==> Installing arpack-ng
+  ==> Using cached archive: /usr/local/var/spack/cache/arpack-ng/arpack-ng-3.5.0.tar.gz
+  ==> Already staged arpack-ng-3.5.0-bloz7cqirpdxj33pg7uj32zs5likz2un in /usr/local/var/spack/stage/arpack-ng-3.5.0-bloz7cqirpdxj33pg7uj32zs5likz2un
+  ==> No patches needed for arpack-ng
+  ==> Building arpack-ng [Package]
+  ==> Executing phase: 'install'
+  ==> Error: RuntimeError: Unable to recursively locate netlib-lapack libraries in /usr/local/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/netlib-lapack-3.6.1-jjfe23wgt7nkjnp2adeklhseg3ftpx6z
+  RuntimeError: RuntimeError: Unable to recursively locate netlib-lapack libraries in /usr/local/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/netlib-lapack-3.6.1-jjfe23wgt7nkjnp2adeklhseg3ftpx6z
+
+  /usr/local/var/spack/repos/builtin/packages/arpack-ng/package.py:105, in install:
+       5             options.append('-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib' % prefix)
+       6
+       7             # Make sure we use Spack's blas/lapack:
+    >> 8             lapack_libs = spec['lapack'].libs.joined(';')
+       9             blas_libs = spec['blas'].libs.joined(';')
+       10
+       11            options.extend([
+
+  See build log for details:
+    /usr/local/var/spack/stage/arpack-ng-3.5.0-bloz7cqirpdxj33pg7uj32zs5likz2un/arpack-ng-3.5.0/spack-build.out
+
+This is because ``netlib-lapack`` requires extra work, if compared to ``openblas``,
+to expose its build information to other packages. Let's edit it:
+
+.. code-block:: console
+
+  root@advanced-packaging-tutorial:/# spack edit netlib-lapack
+
+and follow the instructions as before. What we need to implement is:
+
+.. code-block:: python
+
+  @property
+  def lapack_libs(self):
+      shared = True if '+shared' in self.spec else False
+      return find_libraries(
+          'liblapack', root=self.prefix, shared=shared, recurse=True
+      )
+
+i.e. a property that returns the correct list of libraries for the LAPACK interface.
```

Comment:
maybe mention that `blas_libs()` is already implemented in this package (or the it's state in this branch).